graph/src/data/store/scalar/bigint.rs

```diff
@@ -207,14 +208,15 @@ impl BigInt {
         }
     }
 
-    pub fn to_unsigned_u256(&self) -> U256 {
+    pub fn to_unsigned_u256(&self) -> Result<U256, anyhow::Error> {
         let (sign, bytes) = self.to_bytes_le();
-        assert!(
-            sign == BigIntSign::NoSign || sign == BigIntSign::Plus,
-            "negative value encountered for U256: {}",
-            self
-        );
-        U256::from_little_endian(&bytes)
+        if sign != BigIntSign::Plus && sign != BigIntSign::NoSign {
```

Comment:
A `Minus` variant is available. It may be clearer to use it

```suggestion
        if BigIntSign::Minus {
```